File: Lab02-ErrorEstimation/lab2.py
from math import asin

def factorial(n):
    if n == 0:
        return 1
    else:
        return n*factorial(n-1)
    

def szereg(x,liczbaOkresow):
    wynik = 0
    for n in range(liczbaOkresow):
        licznik = factorial(2*n)
        mianownik = (4**n)*(factorial(n)**2)*(2*n+1)
        skladnik = (licznik/mianownik)*(x**(2*n+1))
        wynik += skladnik
    return wynik


print("Obliczenie arcsin(x) przy użyciu Szeregu Maclaurina")
x = float(input("Podaj wartosc x:"))
liczbaOkresow = int(input("Podaj liczbe wyrazow szeregu:"))

wartoscPrzyblizona = szereg(x,liczbaOkresow)
wartoscDokladna = asin(x)
blad = abs(wartoscPrzyblizona-wartoscDokladna)

print(f"arcsin(x) przyblizony(N = {liczbaOkresow}):{wartoscPrzyblizona:.10f}")
print(f"arcsin(x) dokladny(asin): {wartoscDokladna:.10f}")
print(f"Blad bezwzgledny: {blad:.2e}")

File: Lab02-ErrorEstimation/lab2.m
function szereg()
    disp('Obliczanie arcsin(x) za pomoc¹ szeregu Maclaurina');
    x = input('Podaj wartoœæ x: ');
    liczbaOkresow = input('Podaj liczbê wyrazów szeregu: ');
    
    wynik = 0;

    for n = 0:(liczbaOkresow - 1)
        licznik = factorial(2 * n);
        mianownik = (4^n) * (factorial(n)^2) * (2 * n + 1);
        skladnik = (licznik / mianownik) * (x^(2 * n + 1));
        wynik = wynik + skladnik;
    end

    disp(['arcsin(x) przybli¿ony (N = ', num2str(liczbaOkresow), '): ', num2str(wynik)]);

    % Sprawdzenie

    wartoscDokladna = asin(x);
    blad = abs(wynik - wartoscDokladna);
    disp(['arcsin(x) dok³adny (asin): ', num2str(wartoscDokladna)]);
    disp(['B³¹d bezwzglêdny: ', num2str(blad)]);
end

